src/test/infra-services.ts

```diff
@@ -0,0 +1,332 @@
+                expect(staticMocksBlockstackNamingServiceApiClient.getNameDetails.calledWith(sinon.match.string, cruxdevBlockstackName)).to.be.true;
+            })
+            it('"testcase should return the status availability"', async () => {
+                staticMocksBlockstackNamingServiceApiClient.getNameDetails.resolves(testcaseNameDetail);
+                const nameDetails = await blockstackService.getNameDetails(testcaseCruxDomainId);
+                expect(nameDetails).haveOwnProperty("status").to.be.equal("available");
+                expect(staticMocksBlockstackNamingServiceApiClient.getNameDetails.calledTwice).to.be.true;
+                expect(staticMocksBlockstackNamingServiceApiClient.getNameDetails.calledWith(sinon.match.string, testcaseBlockstackName)).to.be.true;
+            })
+            it('"pendingId" should return with "more" field')
+        })
+        describe('getGaiaHub tests', () => {
+            it('"cruxdev" should return cruxpay gaia hub', async () => {
+                staticMocksBlockstackNamingServiceApiClient.getNameDetails.resolves(cruxdevConfigNameDetails);
+                const gaiaHub = await blockstackService.getGaiaHub(cruxdevConfigCruxId);
+                expect(gaiaHub).to.be.equal(cruxGaiaHub);
+                expect(staticMocksBlockstackNamingServiceApiClient.getNameDetails.calledTwice).to.be.true;
+                expect(staticMocksBlockstackNamingServiceApiClient.getNameDetails.calledWith(sinon.match.string, cruxdevConfigBlockstackName)).to.be.true;
+            })
+        })
+        describe('getDomainRegistrationStatus tests', () => {
+            it('"cruxdev" should be REGISTERED', async () => {
+                staticMocksBlockstackNamingServiceApiClient.getNameDetails.resolves(cruxdevNameDetail);
+                const domainAvailability = await blockstackService.getDomainRegistrationStatus(cruxdevDomainId);
+                expect(domainAvailability).to.be.equal(DomainRegistrationStatus.REGISTERED);
+                expect(staticMocksBlockstackNamingServiceApiClient.getNameDetails.calledTwice).to.be.true;
+                expect(staticMocksBlockstackNamingServiceApiClient.getNameDetails.calledWith(sinon.match.string, cruxdevBlockstackName)).to.be.true;
+            }) 
+            it('"testcase" should be AVAILABLE', async () => {
+                staticMocksBlockstackNamingServiceApiClient.getNameDetails.resolves(testcaseNameDetail);
+                const domainAvailability = await blockstackService.getDomainRegistrationStatus(testcaseCruxDomainId);
+                expect(domainAvailability).to.be.equal(DomainRegistrationStatus.AVAILABLE);
+                expect(staticMocksBlockstackNamingServiceApiClient.getNameDetails.calledTwice).to.be.true;
+                expect(staticMocksBlockstackNamingServiceApiClient.getNameDetails.calledWith(sinon.match.string, testcaseBlockstackName)).to.be.true;
+            })
+            it('"pendingId" should be PENDING')
```

Comment:
seems like a pending test case.

Reply:
removed this anticipated case which will never be possible.

Reply:
https://github.com/blockstack/blockstack-core/blob/d4616af86aeb0f9086c3c32fb62831e85a69ee1a/blockstack/lib/rpc.py#L557

Reply:
https://github.com/cruxprotocol/js-sdk/pull/152/commits/13850a2b3fa7303aa380b110733a7eceac37f6a4